drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/esg_funding_assistance_loader.rb

```diff
@@ -0,0 +1,108 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# matriculation to new platform
+# creates CustomService and CustomDataElements
+module HmisExternalApis::AcHmis::Importers::Loaders
+  class EsgFundingAssistanceLoader < SingleFileLoader
+    def perform
+      records = build_records
+      # destroy existing records and re-import
+      model_class.where(data_source: data_source).destroy_all if clobber
+
+      # can't do bulk insert here since polymorphic CDE's don't seem to work
+      # and bulk-insert returned ids are not ordered
+      without_paper_trail do
+        records.each { |record| record.save!(validate: false) }
+      end
+      Rails.logger.info "#{self.class.name} inserted: #{records.size} records into #{model_class.table_name}"
+    end
+
+    protected
+
+    def filename
+      'ESGFundingAssistance.csv'
+    end
+
+    def build_records
+      personal_id_by_enrollment_id = Hmis::Hud::Enrollment
+        .where(data_source: data_source)
+        .pluck(:enrollment_id, :personal_id)
+        .to_h
+
+      rows.map do |row|
+        enrollment_id = row_value(row, field: 'ENROLLMENTID')
```

Comment:
Again, should we pass `required` so that we error here rather than at validation